### DISCUSSION TITLE
Gulp com método POST

### DISCUSSION THREAD

**Discussion by didiraja** [frontendbr/forum#882] [Perguntas e Respostas]
Pessoal, to começando a desvendar **NPM/Gulp** e tá sendo uma maravilha, realmente essas ferramentas são ótimas e tão adiantando muito o trabalho.

Tô chegando bem perto de montar um ambiente local mais intuitivo que o XAMP. Como trabalhamos com formulário, preciso de um servidor PHP e que faça requisição POST. Porém, mesmo montando servidor usando **gulp-connect-php7**, recebi o erro 405. _(o erro está dando no jquery porque fazemos o envio do formulário via ajax)_

![image](https://user-images.githubusercontent.com/10463976/34739361-3012f198-f563-11e7-8a0c-f79dc9d6b8c0.png)

Procurei no Google e não achei solução pra isso, apenas outras pessoas tendo o mesmo problema. Existe alguma maneira do **gulp fazer servidor local com POST**?

Segue o código do meu gulpfile:

```
var gulp = require('gulp'),
    connect = require('gulp-connect-php7'),
    browserSync = require('browser-sync')/*.create()*/;

// Localhost PHP & BrowserSync
gulp.task('server-php', function() {
    
    connect.server({
        base: 'src/'
    },
        function (){ browserSync({
          /*proxy: '127.0.0.1:8000',*/
            server: {
                baseDir: "src/"
            }
        }); });
    
    gulp.watch("src/**").on('change', browserSync.reload);

}); 
```

**Comment by felipefialho** [comment]
Conseguiu resolver @didiraja ?

**Comment by felipemfp** [comment]
Cara, será que tu não tá mandando o POST para porta errada? O padrão é a porta 8000, não?

**Comment by didiraja** [comment]
@LFeh Cacei na net e nada. O que me leva a crer que esses plugins, no maximo abrem .php e exibem os includes mas nem testei muito disso não. O jeito tá sendo usar o XAMPP só quando preciso testar o formulário.

@felipemfp Dá pra explicar melhor? Sou bem iniciante em back. Só sei dizer, que no XAMPP eu não preciso configurar o post separadamente. Ele ja funciona na mesma porta do localhost.
